### PR TITLE
feat(forks,tests): Osaka

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
- All references to `PragueEIP7692` have been renamed to `Osaka`, including the fork class
- `CancunEIP7692` still exists but a comment has been added to signal deprecation
- `tests/prague/eip7692_eof_v1` has been moved to `tests/osaka` and references have been updated

## 🔗 Related Issues
None

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: tests/osaka/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+"""
+Test cases for EVM functionality introduced in Osaka, EIP-XXXX: Hardfork Meta - Osaka (Pending).
```
Comment (shemnon) [inline tests/osaka/__init__.py:2]:
```suggestion
Test cases for EVM functionality introduced in Osaka, [EIP-7607: Hardfork Meta - Fusaka](https://eip.directory/eips/eip-7607).
```

File: src/ethereum_test_forks/forks/forks.py [hunk 2]
```diff
@@ -1000,22 +1000,17 @@ def solc_min_version(cls) -> Version:
         return Version.parse("1.0.0")  # set a high version; currently unknown
 
 
-class PragueEIP7692(  # noqa: SC200
-    Prague,
-    transition_tool_name="Prague",  # Besu enables EOF at Prague
-    blockchain_test_network_name="Prague",  # Besu enables EOF at Prague
-    solc_name="cancun",
-):
+class Osaka(Prague, solc_name="cancun"):
```
Comment (shemnon) [inline src/ethereum_test_forks/forks/forks.py:1003]: Should this be added to the __all__ section in `__init__.py` for this package?

Comment (marioevz) [reply src/ethereum_test_forks/forks/forks.py:1003]: Indeed, thanks!

File: tests/osaka/eip7692_eof_v1/eof_tracker.md [hunk 3]
```diff
@@ -1,83 +1,83 @@
...
+- [x] Multiple data section headers ([`tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py`](./eip3540_eof_v1/test_container_validation/index.md)`-k multiple_data_sections`)
+- [x] Container without type section ([`tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py::test_section_order`](./eip3540_eof_v1/test_section_order/index.md) `-k 'SectionTest.MISSING-section_kind_TYPE'`)
+- [x] Container without code sections ([`tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py::test_section_order`](./eip3540_eof_v1/test_section_order/index.md) `-k 'SectionTest.MISSING-section_kind_CODE'`)
+- [x] Container without data section ([`tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py::test_section_order`](./eip3540_eof_v1/test_section_order/index.md) `-k 'SectionTest.MISSING-section_kind_DATA'`)
+- [x] Valid containers without data section and with subcontainers ([`tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_valid_containers`](./eip3540_eof_v1/test_container_validation/test_valid_containers.md)`[fork_CancunEIP7692-eof_test-single_subcontainer_without_data]`)
```
Comment (danceratopz) [inline tests/osaka/eip7692_eof_v1/eof_tracker.md:66]: Can these test case IDs be updated to `Osaka` yet? Do they still need to be filled with `evm-one`?
```suggestion
- [x] Valid containers without data section and with subcontainers ([`tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_container_validation.py::test_valid_containers`](./eip3540_eof_v1/test_container_validation/test_valid_containers.md)`[fork_Osaka-eof_test-single_subcontainer_without_data]`)
```

**Review by danceratopz — APPROVED** [review]
LGTM! Few minor suggestions and a question.

**Pull request merged**